gateway/ln-gateway/src/lib.rs

```diff
@@ -1349,6 +1338,32 @@ impl Gateway {
 
         Ok(())
     }
+
+    pub async fn execute_with_lightning_connection<F, Fut, T>(&self, func: F) -> anyhow::Result<T>
+    where
+        F: FnOnce(Arc<dyn ILnRpcClient>, PublicKey, String, Network) -> Fut,
+        Fut: Future<Output = anyhow::Result<T>>,
+    {
+        match self.state.read().await.clone() {
+            GatewayState::Running {
+                lnrpc,
+                lightning_public_key,
+                lightning_alias,
+                lightning_network,
+            } => {
+                func(
+                    lnrpc,
+                    lightning_public_key,
+                    lightning_alias,
+                    lightning_network,
+                )
+                .await
+            }
+            _ => {
+                anyhow::bail!("Lightning node is not connected");
+            }
+        }
+    }
```

Comment:
This seems overly complicated, why not have a function that returns a `Result<RunningContext, anyhow::Error>` where `RunningContext` is the 4 fn arguments? Then that could be `?`-ed and used and result in the same semantics: early return with error if not running.

Reply:
Replaced `execute_with_lightning_connection` with `get_lightning_context`